### PR TITLE
DEP-2203: Keda enable TLS

### PR DESCRIPTION
**Pull request by ella-ktech**

### CODE REVIEW

File: charts/bandstand-headless-service/templates/scaled-object.yaml [hunk 1]
```diff
@@ -36,5 +36,10 @@ spec:
         value: {{ quote (.threshold | default 75) }}
       metricType: Utilization
     {{- end }}
+    {{- if eq .type "kafka" }}
+    - type: kafka
```
Comment (gavinclarkeuk) [inline charts/bandstand-headless-service/templates/scaled-object.yaml:40]: I obviously missed adding kafka to this chart, it should be identical to the web services chart. Can just copy the whole thing over

Comment (ella-ktech) [reply charts/bandstand-headless-service/templates/scaled-object.yaml:40]: Thanks

**Comment by gavinclarkeuk** [comment]
Looks good. Can you add the field to the docs for each chart, including the valid values (from the keda docs). Also might want to make the test case use a valid value to avoid any confusion.

File: charts/bandstand-web-service/README.md [hunk 2]
```diff
@@ -37,8 +37,12 @@
...
+| kafka.topic                  | string |                                                            | Name of the topic on which processing the offset lag. lag.                                                                                                                                                                                                                                        |
+| kafka.tls                  | string | `disable`                                                           | To enable SSL auth for Kafka, set this to enable. If not set, TLS for Kafka is not used.                                                                                                                                                                                                                                        |
```
Comment (gavinclarkeuk) [inline charts/bandstand-web-service/README.md:44]: The chart seems to defaulting this to 'enable'. I'm ok with it being enabled by default, but the docs should match (don't forget the headless-svc docs too)

File: test-charts/headless-service/advanced-scaling/values.yaml [hunk 3]
```diff
@@ -24,3 +24,8 @@ bandstand-web-service:
         desiredReplicas: 1
         start: 0 7 * * MON-FRI
         end: 0 19 * * MON-FRI
+      - type: kafka
+        topic: event.msk.reference.dummy
+        consumerGroup: keda-consumer
+        bootstrapServersEnvVar: MSK_BOOTSTRAP_SERVERS # optional - defaults to KAFKA_BROKERS
+        tls: enabled
```
Comment (gavinclarkeuk) [inline test-charts/headless-service/advanced-scaling/values.yaml:31]: Whilst it doesn't really matter for the purpose of the test would be better if these examples used valid values 'enable' or 'disable'

**Comment by gavinclarkeuk** [comment]
Overall this is looking good. If you can fix the couple of minor things we'll get this merged, then would be great if you could follow up with Anna on the thread from the help channel to say this is available.

I don't mind if the default is enable or disable, but they will need to know which we've gone for.

**Comment by ella-ktech** [comment]
@gavinclarkeuk thank you, changed now. Defaults to "enable" & put "enable" in the test files.

File: charts/bandstand-headless-service/README.md [hunk 4]
```diff
@@ -23,6 +23,10 @@
...
+| kafka.topic                  | string |                                                            | Name of the topic on which processing the offset lag. lag.                                                                                                                                                                                                                                        |
+| kafka.tls                  | string | `enable`                                                           | To enable SSL auth for Kafka, set this to enable. If not set, TLS for Kafka is not used.                                                                                                                                                                                                                                        |
```
Comment (gavinclarkeuk) [inline charts/bandstand-headless-service/README.md:29]:
```suggestion
| kafka.tls                  | string | `enable`                                                           | To enable SSL auth for Kafka. TLS is enabled by default, to turn it off set this to `disable` .                                                                                                                                                                                                                                   |
```

**Review by gavinclarkeuk — APPROVED** [review]

**Pull request merged**